example/speech_recognition/singleton.py

```diff
@@ -32,7 +32,7 @@ def getInstance(self):
 
     def __new__(class_, *args, **kwargs):
         print("__new__")
-        class_.instances[class_] = super(Singleton, class_).__new__(class_, *args, **kwargs)
+        class_.instances[class_] = super().__new__(class_, *args, **kwargs)
```

Comment:
why remove passing `Singleton` and `class_` while calling `super`?

Typo in line 42, should be `SingletonInstance`.
line 52 in redundant.
The whole `SingletonInstance` class defined in line 42 is redundant. It is not being used anywhere. I think it can be removed.

In fact in this whole speech recognition example, the code around the use of singletons can be refactored. The `Singleton` class defined in line 21 is used only once as a decorator in `label_util.py` module. 

The `log_util.py` module again defines another class called `SingletonType` which is used to make `LogUtil` class a singleton. There is duplication of code and logic here.

Reply:
Python3 complained about this usage of super(Singleton, class_).
LogUtil's singleton looks much cleaner. Will refactor the code around this. Wanted to make minimal changes to get the example working/not break it - but I think this is essential. Will make the changes and resubmit. Thanks @anirudhacharya